src/openscad_gui.cc

```diff
@@ -290,6 +294,20 @@ int gui(std::vector<std::string>& inputFiles, const std::filesystem::path& origi
   }
 #endif
 
+#ifdef ENABLE_GUI_TESTS
+  // Adds a singleshot timer that will be executed when the application will be started.
+  // the timer validates taht each mainwindow respects the expected UX behavior.
+  if(gui_test != "none"){
+      QTimer::singleShot(0, [&]()
+      {
+          for(auto w : app.windowManager.getWindows()){
+              runAllTest(w);
```

Comment:
The cmd-line docs say that gui_test should be `all` or a list of tests. I'm fine merging this as WIP, but perhaps add a TODO comment here so it's not forgotten.